### ISSUE TITLE
Persistant Volume - DeleteFileShare fails due to always using data plane API

### ISSUE THREAD

**Issue by kevinharing** [kubernetes-sigs/azurefile-csi-driver#1111]
**What happened**:

A VolumeFailedDelete error is being triggered on all my PVs:
```
message: >-
  rpc error: code = Internal desc = DeleteFileShare
  pvc-42b5d5f9-afe3-4619-a5ee-694e3107915c under account(xxx)
  rg(MC_rg-xxx-aks-01) failed with error:
  storage: service returned error: StatusCode=403,
  ErrorCode=AuthorizationFailure, ErrorMessage=This request is not authorized to
  perform this operation.
source:
  component: >-
    file.csi.azure.com_csi-azurefile-controller-68454b4864-6wpgp_eaccbef6-cbb4-47e7-981f-de485d14735d
```

**What you expected to happen**:
The volumes are deleted

**How to reproduce it**:
I have a storage account (BYO) with a firewall set up that whitelists our AKS subnet. The shares are created and mounted correctly on the pods via CSI, only the removal fails. I suspect that the share is being deleted via the data plane API, but I did not set that property to "true" as specified in the docs. I want it to use the control plane API.

In the storage account logs it shows the CreateShare action completes successfully, but the DeleteShare action fails with a 403/AuthorizationFailure.

**Anything else we need to know?**:
CSI is managed by AKS.

**Environment**:

- CSI Driver version: 1.21.0
- Kubernetes version (use `kubectl version`): 1.24.6

**Comment by andyzhangx** [comment]
hi @kevinharing what's your storage class config? if you don't specify `csi.storage.k8s.io/provisioner-secret-name` in storage class, CSI driver would use control plane API.

**Comment by kevinharing** [comment]
> hi @kevinharing what's your storage class config? if you don't specify `csi.storage.k8s.io/provisioner-secret-name` in storage class, CSI driver would use control plane API.

We don’t specify a custom secret name, so I’d expect it would take the default. Which it does by looking at the secret in the cluster (in the kube-system namespace, which we specify via the parameters in the storage class).

Do you mean that not specifying this parameter triggers this bug?

**Comment by andyzhangx** [comment]
@kevinharing I have figured it out, this PR(https://github.com/kubernetes-sigs/azurefile-csi-driver/pull/1082) should fix the issue, and that PR is in v1.22.0 release which is still under rolling out on AKS, it still needs a few days to roll out complete in your westeuope region.

 - related logs:
```
W1014 22:23:04.222082       1 azurefile.go:761] switch to use data plane API instead for account clusterdevsa since it's throttled
```

**Comment by kevinharing** [comment]
> @kevinharing I have figured it out, this PR(#1082) should fix the issue, and that PR is in v1.22.0 release which is still under rolling out on AKS, it still needs a few days to roll out complete in your westeuope region.
> 
> * related logs:
> 
> ```
> W1014 22:23:04.222082       1 azurefile.go:761] switch to use data plane API instead for account xxx since it's throttled
> ```

Thanks for the quick help. Will wait for this new version to see if it’s fixed!

**Comment by andyzhangx** [comment]
the fix has been rolled out to westeurope region, pls try again.
